ibc-derive/src/utils.rs

```diff
@@ -4,73 +4,109 @@ use syn::{Path, Variant};
 
 /// Encodes the ibc-rs types that will be used in the macro
 ///
-/// Note: we use `ibc` as our top-level crate, due to the
-/// `extern crate ibc as ibc;` statement we inject.
-pub struct Imports;
+/// Note: we use `ibc` or `ibc-core` as our top-level crate, due to the `extern
+/// crate ibc as ibc;` statement we inject.
+pub struct Imports {
+    prefix: TokenStream,
+}
+
+impl Imports {
+    pub fn new_ibc() -> Self {
+        Self {
+            prefix: quote! {::ibc::core},
+        }
+    }
+
+    pub fn new_ibc_core() -> Self {
+        Self {
+            prefix: quote! {::ibc_core},
+        }
+    }
+
+    pub fn prefix(&self) -> &TokenStream {
+        &self.prefix
+    }
+}
```

Comment:
IMO this seems like `ibc-derive` requires `ibc_core`.

Can't we add that in `ibc_derive` dependency and exports and use it here as `ibc_derive::ibc_core` or something like that?

Reply:
Until expansion takes place, the inclusion of `ibc_core` isn't happening. However, I've adjusted things by relocating each set of derive macros to their relevant meta-crate. They are are now placed under the `derive` feature flag. This ensures that when working with `::ibc`, we won't unintentionally import `::ibc_core`, and vice versa. See if you like it?